x/stakeibc/keeper/callbacks.go

```diff
@@ -39,12 +39,12 @@ func (c Callbacks) Has(id string) bool {
 	return found
 }
 
-func (c Callbacks) AddCallback(id string, fn interface{}) types.QueryCallbacks {
+func (c Callbacks) AddCallback(id string, fn interface{}) icqtypes.QueryCallbacks {
 	c.callbacks[id] = fn.(Callback)
 	return c
 }
 
-func (c Callbacks) RegisterCallbacks() types.QueryCallbacks {
+func (c Callbacks) RegisterCallbacks() icqtypes.QueryCallbacks {
```

Comment:
Just to make sure I'm following the flow, it goes:

1. Query withdrawal balance with ICQ (initiated from hooks) which gives us the amount of outstanding rewards
2. When the query returns, ICQ callback `callbacks.go` is executed 
3. The ICQ callback initiates an ICA transfer of 90% of that rewards amount from the withdrawal account to the delegation account
4. When the ICA returns, the Reinvestment callback is executed, which creates a new deposit record so the rewards can be swept up and staked in the next epoch

Is that right?

Reply:
that's right!